straxen/contexts.py

```diff
@@ -218,17 +219,100 @@ def xenonnt_led(**kwargs):
     return st
 
 
-def xenonnt_simulation(output_folder='./strax_data'):
+def xenonnt_simulation(output_folder='./strax_data',
+                       cmt_run_id='020280',
+                       cmt_version='v4',
+                       cmt_overwrite=immutabledict(),
+                       fax_config='fax_config_nt_design.json',
+                       overwrite_from_fax_file=False,
+                       _allow_diverging_settings=False,
+                       **kwargs):
+    """
+    Context with already pre-initialized corrections values for CMT.
+    Sets correction values according cmt_run_id and specified version.
+    Settings can be overwritten via cmt_overwrite or fax config file.
+
+    :param output_folder: Output folder for strax data.
+    :param cmt_run_id: Run id which should be used to load corrections.
+    :param cmt_version: Global version for corrections to be loaded.
+    :param cmt_overwrite: Dictionary to overwrite CMT settings. Keys
+        must be valid CMT option keys.
+    :param fax_config: Fax config file to use.
+    :param overwrite_from_fax_file: If true overwrites CMT settings
+        from fax file.
+    :param _allow_diverging_settings: Disable check for matching CMT
+        settings in context and fax file settings. Experts use only.
+    :param kwargs: Additional kwargs taken by strax.Context.
+
+    :return: strax.Context instance
+    """
     import wfsim
     st = strax.Context(
         storage=strax.DataDirectory(output_folder),
         config=dict(detector='XENONnT',
-                    fax_config='fax_config_nt_design.json',
+                    fax_config=fax_config,
                     check_raw_record_overlaps=True,
-                    **straxen.contexts.xnt_simulation_config,
+                    **straxen.contexts.xnt_common_config,
                     ),
-        **straxen.contexts.xnt_common_opts)
+        **straxen.contexts.xnt_common_opts,  **kwargs)
     st.register(wfsim.RawRecordsFromFaxNT)
+
+    st.apply_cmt_version(f'global_{cmt_version}')
+    if not cmt_run_id:
+        raise ValueError('You have to specify a run_id which should be used to initialize '
+                         'the corrections.')
+
+    cmt_options = straxen.get_corrections.get_cmt_options(st)
+
+    p = st.get_single_plugin(cmt_run_id, 'peak_positions')
+    default_reconstruction_algorithm = p.config['default_reconstruction_algorithm']
```

Comment:
I always thought this was a quite slow operation, as you fully init the plugin but I might be remembering it wrong.

Reply:
Initializing the context took me about 5s. So I think it is fine.

Reply:
I see, perhaps it was only if you are actually loading things like gains that it makes it take a long time.

10.000x faster but completely irrelevant (19.4 ms vs 1.13 µs) would be:
st._plugin_class_registry['peak_positions'].takes_config['default_reconstruction_algorithm'].default